doc/spec/federated_learning.rst

```diff
@@ -0,0 +1,353 @@
+Federated Learning in Microsoft EconML Library

```

Comment:
This documentation has some helpful content but also seems very long, with lots of boilerplate/fluff, most of which is not related to our library.

Please trim it down a lot - we don't need an introduction on how to use Spark in general, for example.

Most importantly, this documentation does not include any reference to the actual classes in our library that implement the functionality.  The "Combining Final Results" section looks completely generic and refers to a whole bunch of techniques that have nothing to do with our library.